adaptdl/adaptdl/torch/data.py

```diff
@@ -239,10 +249,26 @@ def _sync_local_bsz(self):
         else:
             # Autoscale batch size, compute on rank 0 and broadcast.
             speedup_fn = get_speedup_fn()
-            _, local_bsz = speedup_fn(adaptdl.env.num_nodes(),
-                                      adaptdl.env.num_replicas(),
-                                      return_local_bsz=True)
-            self.current_local_bsz = adaptdl.collective.broadcast(local_bsz)
+            speedup, local_bsz = speedup_fn(adaptdl.env.num_nodes(),
+                                            adaptdl.env.num_replicas(),
+                                            return_local_bsz=True)
+
+            # if first time called, initiliza the batch size field
+            if(self.init_time):
+                self.cur_local_bsz = local_bsz
+                self.init_time = False
+
+            # if the speedup is significant, we use it, 
+            # otherwise, keep the old one.
+
+            if(speedup > self.speedup_threshold):
```

Comment:
Doesn't this just accept any batch size with a speedup of 1.05 or greater? What we want is to only change the batch size if the _increase_ in speedup over the current speedup surpasses some threshold.